internal/pkg/deploy/cloudformation/stack/lb_web_svc.go

```diff
@@ -344,15 +344,15 @@ func (s *LoadBalancedWebService) Parameters() ([]*cloudformation.Parameter, erro
 			},
 		}...)
 	}
-	if !s.manifest.NLBConfig.IsEmpty() {
-		port, _, err := manifest.ParsePortMapping(s.manifest.NLBConfig.Port)
+	if !s.manifest.NLBConfig.PrimaryRoutingRule.IsEmpty() {
+		port, _, err := manifest.ParsePortMapping(s.manifest.NLBConfig.PrimaryRoutingRule.Port)
```

Comment:
If you are talking about `NLBAliases` and `NLBPort` template params then we do need them in uri.go to build our NLB links that customer will use to send their request. I know it's not the most perfect way to do it, but that's how we have done in the past. Me and Penghao agreed on the point that we should get rid of that logic to use template params to describe NLB links. But we will do it separately in general refactoring instead of multiple port feature. Then we can remove these params.  Does that sound good to you?